package/src/components/ChannelPreview/hooks/useChannelPreviewData.ts

```diff
@@ -70,16 +97,35 @@ export const useChannelPreviewData = <
       refreshUnreadCount();
     };
 
+    const handleNewMessageEvent = (event: Event<StreamChatGenerics>) => {
```

Comment:
These needed to be brought back because of the `messageSets` manipulation we're doing in the RN SDK. Reading the data directly from `channel.state.latestMessages` is not always reliable as we change this on the fly. Once that logic is removed, perhaps we can revisit this. 

Even so, `channel.state.latestMessages` works in `O(n)` time in terms of the number of `messageSets` that we have so we have to be careful here or at the very least do it in a smart way.